components/ruby/lib/chef-licensing/config.rb

```diff
@@ -47,6 +63,19 @@ def license_add_command
       def license_list_command
         @license_list_command ||= "license list"
       end
+
+      private
+
+      def log_level_flags_present?
+        !get_log_level_from_flags.nil?
+      end
+
+      def get_log_level_from_flags
+        ChefLicensing::ArgFetcher.fetch_value("--log-level", :string) ||
+          ChefLicensing::ArgFetcher.fetch_value("--chef-log-level", :string) ||
+          ChefLicensing::EnvFetcher.fetch_value("LOG_LEVEL", :string) ||
+          ChefLicensing::EnvFetcher.fetch_value("CHEF_LOG_LEVEL", :string)
```

Comment:
what is the right precedence here? my understand is ENV variables override the options provided

Reply:
No, we give precedence to arguments over ENV-based values. Check the existing implementations for other keys as well in this library